druntime/src/rt/cast_.d

```diff
@@ -23,10 +23,11 @@ pure:
 // but we are trying to implement dynamic cast.
 extern (D) private bool areClassInfosEqual(scope const ClassInfo a, scope const ClassInfo b) @safe
 {
-    if (a is b)
-        return true;
-    // take care of potential duplicates across binaries
-    return a.name == b.name;
+    // same class if signatures match, works with potential duplicates across binaries
+    return a.m_flags & TypeInfo_Class.ClassFlags.hasNameSig
```

Comment:
Nit: Would be cleaner to do the old way in an if statement that returns early.

Reply:
The new way requires two comparisons max. The old way requires one and then a string comparison. I want to get rid of the old way, but @ibuclaw suggested keeping it for a release or two.

The reason for the change is deadalnix has identified it as serious performance bottleneck.

Reply:
I gathered (I've discussed this amaury in person), it just reads nicer to do not do the whole shebang in the same expression (this is my main gripe with a lot of DMD, too many subexpressions your majesty!)